va/va.h

```diff
@@ -3076,6 +3076,10 @@ typedef struct _VAPictureParameterBufferH264
         uint32_t value;
     } pic_fields;
     uint16_t frame_num;
+    /** \brief Same as the H.264 bitstream syntax element. */
+    uint8_t num_ref_idx_l0_default_active_minus1;
+    /** \brief Same as the H.264 bitstream syntax element. */
+    uint8_t num_ref_idx_l1_default_active_minus1;
```

Comment:
Hantro/VSI hardware works at frame level. For my own understanding, how do one derive from slices this information ?

Reply:
from syntax level, num_ref_idx_lN_active_minus1 will be override by slice parameter (depend on whether num_ref_idx_active_override_flag is true). so I dont think that you just need a frame level syntax for it.  if there are multiple slice with different values. how to handle it?

Reply:
Ok, I'm not VSI employee, so I can only guess thing, let me know if that make sense. The HW will be parsing the value from the bitstream by itself, ignoring the values we have in VA-API structure per slice. Now, if it encounter a slice that does not have this syntax, it needs to use the default value, which we are adding here.

My question is, can we decuce from the information already in VA-API what is the default value ? Is it the maximum of all slices or something like this ?

Reply:
if application parse the bitstream, it always know whether the default value should be overridden.  it's why we just use slice parameter to set the value no matter it use default value or value from slice. it could cover all of them.  

so, I assume your problem is: application just parse part of the bitstream, SPS, PPS. then send other remaining bitstream to HW. HW may handle the sliceheader parser, and check whether it should be overridden. right?

for this case, the easiest way is sending a slice parameter, which just use to send these two parameters . other parameters in the slice parameter will be ignored.

Reply:
Said like this is sounds more like a hack, the application have no idea if the internal is slice or frame based. If you set the default in the firs slice, and the first slice wasn't using the defaults, you endup with the wrong value. I don't really know what this one is for and the semantic of it internally, that's a bit why I ask what to with only the squashed version (per slice final result). I think these two are somewhat legit request. But I guess it does not matter, since the driver would need reparse anyway to figure-out the skip bits parameters. I think the double parse overhead is unavoidable in the context of VA-API.with this type of HW.